client/src/modules/list/components/Items/model/actions.js

```diff
@@ -0,0 +1,190 @@
+import { ENDPOINT_URL } from 'common/constants/variables';
+import { patchData, postData } from 'common/utils/fetchMethods';
+import { ItemActionTypes } from 'modules/list/components/Items/model/actionTypes';
+import { MessageType as NotificationType } from 'common/constants/enums';
+import { createNotificationWithTimeout } from 'modules/notification/model/actions';
+
+const addItemFailure = errorMessage => ({
+  type: ItemActionTypes.ADD_FAILURE,
+  payload: errorMessage
+});
+const addItemSuccess = (item, listId) => ({
+  type: ItemActionTypes.ADD_SUCCESS,
+  payload: { item, listId }
+});
+const addItemRequest = () => ({
+  type: ItemActionTypes.ADD_REQUEST
+});
+
+const toggleItemSuccess = (item, listId) => ({
+  type: ItemActionTypes.TOGGLE_SUCCESS,
+  payload: { item, listId }
+});
+const toggleItemRequest = () => ({
+  type: ItemActionTypes.TOGGLE_REQUEST
+});
+const toggleItemFailure = errMessage => ({
+  type: ItemActionTypes.TOGGLE_FAILURE,
+  payload: errMessage
+});
+
+const voteForItemSuccess = (item, listId) => ({
+  type: ItemActionTypes.VOTE_SUCCESS,
+  payload: { item, listId }
+});
+const voteForItemRequest = () => ({
+  type: ItemActionTypes.VOTE_REQUEST
+});
+const voteForItemFailure = errMessage => ({
+  type: ItemActionTypes.VOTE_FAILURE,
+  payload: errMessage
+});
+
+const updateItemDetailsSuccess = (listId, itemId, data) => ({
+  type: ItemActionTypes.UPDATE_DETAILS_SUCCESS,
+  payload: { listId, itemId, data }
+});
+
+const updateItemDetailsRequest = () => ({
+  type: ItemActionTypes.UPDATE_DETAILS_REQUEST
+});
+
+const updateItemDetailsFailure = () => ({
+  type: ItemActionTypes.UPDATE_DETAILS_FAILURE
+});
+
+const cloneItemSuccess = (listId, item) => ({
+  type: ItemActionTypes.CLONE_SUCCESS,
+  payload: { listId, item }
+});
+
+const cloneItemRequest = () => ({
+  type: ItemActionTypes.CLONE_REQUEST
+});
+
+const cloneItemFailure = () => ({
+  type: ItemActionTypes.CLONE_FAILURE
+});
+
+export const addItem = (item, listId) => dispatch => {
+  dispatch(addItemRequest());
+  return postData(`${ENDPOINT_URL}/lists/add-item`, { item, listId })
+    .then(resp => resp.json())
+    .then(json => dispatch(addItemSuccess(json, listId)))
+    .catch(err => {
+      dispatch(addItemFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message || "Oops, we're sorry, adding item failed..."
+      );
+    });
+};
+
+export const toggle = (
+  isOrdered,
+  itemId,
+  listId,
+  updatedAuthor
+) => dispatch => {
+  dispatch(toggleItemRequest());
+  return patchData(`${ENDPOINT_URL}/lists/${listId}/update-item`, {
+    authorName: updatedAuthor,
+    itemId,
+    isOrdered: !isOrdered,
+    listId
```

Comment:
do you need to pass all that data to update `isOrdered`?
`itemId` and `isOrdered` should be enough - `listId` is already in api path

Reply:
I removed "listId". I removed updatedAuthor from passing it to backend (it's not used there now) but I left it in action. This "toggling item" functionality will be refactored in the future task "EOC-151 Fix item toggling (restoring should change author)" because for now it only changes "isOrdered" field. We should fix it to store new User Id when an item is restored, so "updatedAuthor" variable will be needed then.